src/fenwick.jl

```diff
@@ -0,0 +1,95 @@
+"""
+    FenwickTree{T}(n)
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) of length `n`.
+ 
+"""
+mutable struct FenwickTree{T}
```

Comment:
```suggestion
struct FenwickTree{T}
```

One more suggested change: let's make the struct immutable.  This still allows the `bi_tree` to be modified, but it won't allow `n` to change.